packages/botonic-plugin-flow-builder/src/action.tsx

```diff
@@ -75,3 +70,30 @@ export class FlowBuilderMultichannelAction extends FlowBuilderAction {
     )
   }
 }
+
+function getNodeByUserInput(
+  cmsApi: FlowBuilderApi,
+  locale: string,
+  request: ActionRequest
+): HtNodeWithContent | undefined {
+  if (request.session.is_first_interaction) {
+    const startNode = cmsApi.getStartNode()
+    return startNode
+  }
+
+  if (request.input.data) {
+    const intentNode = cmsApi.getNodeByIntent(request.input, locale)
+    const keywordNode = cmsApi.getNodeByKeyword(request.input.data, locale)
+    return intentNode ?? keywordNode
```

Comment:
if `intentNode` is defined we could return it before computing the `keywordNode`

Reply:
yep, this is a symptom that we are facing an issue with typing somewhere